framework/src/materials/DerivativeSumMaterial.C

```diff
@@ -105,8 +108,9 @@ DerivativeSumMaterial::DerivativeSumMaterial(const InputParameters & parameters)
 void
 DerivativeSumMaterial::initialSetup()
 {
-  for (unsigned int n = 0; n < _num_materials; ++n)
-    validateCoupling<Real>(_sum_materials[n]);
+  if (_validate_coupling)
+    for (unsigned int n = 0; n < _num_materials; ++n)
+      validateCoupling<Real>(_sum_materials[n]);
```

Comment:
:+1: I think this is the best way of doing it. It allows users to set a GlobalParam if they want to skip this check.